monkey/monkey_island/cc/resources/auth/request_otp.py

```diff
@@ -0,0 +1,23 @@
+from flask import make_response
+
+from monkey_island.cc.flask_utils import AbstractResource
+
+
+class RequestOTP(AbstractResource):
+    """
+    A resource for requesting a one-time password
+
+    One-time passwords may be requested by an Agent that has already authenticated,
+    so that Agents that it propagates can register.
+    """
+
+    urls = ["/api/request-otp"]
```

Comment:
A URL like `new-otp` might be more "RESTful".

Reply:
I feel like just `otp` is enough.

Reply:
I think this should be `/api/agent-otp` to be consistent with the current interface. This OTP is specifically for agents. If we wanted to have MFA for user registration, that resource would be `/api/otp`